service/internal/checksum/middleware_compute_input_checksum.go

```diff
@@ -268,6 +264,10 @@ func (m *addInputChecksumTrailer) HandleFinalize(
 ) (
 	out middleware.FinalizeOutput, metadata middleware.Metadata, err error,
 ) {
+	v, ok := ctx.Value(useTrailer{}).(bool)
+	if !ok || !v {
+		return next.HandleFinalize(ctx, in)
+	}
```

Comment:
Overcomplicated:

```suggestion
	if enabled, _ := ctx.Value(useTrailer{}).(bool); enabled {
		return next.HandleFinalize(ctx, in)
	}
```

(presence doesn't matter, false is false)